### PR TITLE
Adding customJsPath options

### PR DESCRIPTION
**Pull request by roggervalf**
This PR is for customizing the layout by providing a new feature: customJsPath, so it could be pass as a param. In order to test this PR:
1. Go to examples
2. Run: npm run start:bull
3. You should see a google link in nav bar at the button
Selection_022

### CODE REVIEW

**Comment by roggervalf** [comment]
@bradvogel please when you get a chance, some :eyes: in this PR

File: example/bull.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ async function main() {
     },
     {
       port: HTTP_SERVER_PORT,
+      externalLinks: [{ link: 'https://www.google.com/', text: 'google' }],
```
Comment (roggervalf) [inline example/bull.js:60]: I added it for now, just to see in it with an example, after this PR is checked I could delete it

File: src/server/app.js [hunk 2]
```diff
@@ -23,7 +23,6 @@ module.exports = function (config) {
   app.locals.Queues = queues;
   app.locals.appBasePath = '';
   app.locals.vendorPath = '/vendor';
-  app.locals.customCssPath = config.customCssPath;
```
Comment (roggervalf) [inline src/server/app.js:26]: From docs said that this should come from listenOpts param, not from config that is for queue configuration

File: README.md [hunk 3]
```diff
@@ -167,6 +173,7 @@ router.use('/', arena);
 - `disableListen` - don't let the server listen (useful when mounting Arena as a sub-app of another Express app) (default: false)
 - `useCdn` - set false to use the bundled js and css files (default: true)
 - `customCssPath` - an URL to an external stylesheet (default: null)
+- `externalLinks` - an array of external links to be included in nav (default: null)
```
Comment (bradvogel) [inline README.md:176]: Can you share more about your use case for this? I'm concerned this is a "slippery slope" and there will be an explosion of different parameters to try to customize the UX. What we might need is some way to override the internal HTML templates that Arena uses.

Comment (roggervalf) [reply README.md:176]: sure, in my team we have some other projects related to queues that we would like to add in this nav for easy access when navigating between projects, could it be possible to have this in or something that I can change?

File: README.md [hunk 4]
```diff
@@ -27,31 +27,37 @@ const Arena = require('bull-arena');
...
-  customCssPath: 'https://example.com/custom-arena-styles.css',
-});
+        // Queue key prefix. Defaults to "bq" for Bee and "bull" for Bull.
+        prefix: 'foo',
+      },
+    ],
+  },
+  {
+    // Optionally include your own stylesheet
+    customCssPath: 'https://example.com/custom-arena-styles.css',
```
[the remark this answers is not in the document]
Comment (roggervalf) [reply README.md:55]: I have a doubt here, in readme appears:
Selection_018
that customCssPath should be in listenOpts, not in queue config param :thinking: maybe also that part should be changed too :thinking:

[the remark this answers is not in the document]
Comment (bradvogel) [reply README.md:55]: Great catch. That was another bug in the readme. I fixed via a force push to https://github.com/bee-queue/arena/pull/359/files

File: README.md (hunk 3 above)
Comment (bradvogel) [reply README.md:176]: What if we added a `customJsPath` parameter and you used it to write the links in using client-side JS? My thinking is that a `customJsPath` parameter would provide a lot more flexibility to customize anything.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (roggervalf) [reply README.md:55]: nice let me revert these related changes

File: README.md (hunk 3 above)
Comment (roggervalf) [reply README.md]: Sure let me change this

**Comment by roggervalf** [comment]
in this PR I tried to handle the styled error with pr template https://github.com/bee-queue/arena/pull/353, this is why is failing

File: example/bull.js [hunk 5]
```diff
@@ -54,11 +56,14 @@ async function main() {
           },
         },
       ],
+      customJsPath: 'example.js',
```
Comment (bradvogel) [inline example/bull.js:59]: When I run the Bull example (via `npm run start:bull` from the examples dir) and navigate into a queue, I get the following 404 error:

image

File: README.md [hunk 6]
```diff
@@ -51,6 +51,9 @@ Arena({
 
   // Optionally include your own stylesheet
   customCssPath: 'https://example.com/custom-arena-styles.css',
+
+  // Optionally include your own script
+  customJsPath: 'https://example.com/custom-arena-js.js',
```
Comment (bradvogel) [inline README.md:56]: looks like there's a merge conflict with master

**Review by bradvogel — CHANGES_REQUESTED** [review]

File: example/bull.js (hunk 5 above)
Comment (roggervalf) [reply example/bull.js]: Good catch, I fixed it in last commit

File: README.md (hunk 6 above)
Comment (roggervalf) [reply README.md:56]: I could fixed in last commits. Tyty

**Review by bradvogel — APPROVED** [review]

**Pull request merged**